response_operations_ui/views/collection_exercise.py

```diff
@@ -588,7 +588,7 @@ def edit_collection_exercise_details(short_name, period):
 
 @collection_exercise_bp.route("/<survey_ref>/<short_name>/create-collection-exercise", methods=["GET"])
 @login_required
-def get_create_collection_exercise_form(survey_ref, short_name):
+def get_create_collection_exercise_form(survey_ref, short_name, previous_period=None):
```

Comment:
I think this carries on from my previous comment.  I think function variables are meant to be used for parts of the url (like survey_ref and short_name here).  If you want to get the previous_period from the url parameters, removing this and adding `request.args.get('previous_period')` here would be better